app/Http/Controllers/FlightController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Flight;
+use App\Http\Requests\FlightRequest;
+use App\Http\Resources\FlightResource;
+use App\Jobs\GetFlightAdvisories;
+use App\Jobs\GetFlightWeather;
+use Illuminate\Http\Request;
+
+class FlightController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @param \Illuminate\Http\Request   $request
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+        $flights = Flight::query()
+            ->with('airspace', 'weather')
+            ->paginate();
+
+        if ($request->expectsJson()) {
+            return FlightResource::collection($flights);
+        }
+
+        // handle view
+        return view('flight.index', ['flights' => $flights]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('flight.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \App\Http\Requests\FlightRequest $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(FlightRequest $request)
+    {
+        $request->validate([
```

Comment:
Laravel requests

Reply:
In this case, I used a common Request object (FlightRequest) and added additional rules in the controller as a second validation step that are specific to the Create method, since create and update's only differences were the required validators.

If authentication was in play, or te here was any more differences, they would definitely be separate Request objects.